WordPress/gutenberg
# PrePublishPanel: suggest tags and postformat

Related https://github.com/WordPress/gutenberg/issues/7426

This PR aims to include the Tags and the PostFormat components in the pre-publish panel:

* Tags component is shown only if the user hasn't tagged the post yet.
* PostFormat is shown if the suggested PostFormat is different from the current one.

### How to test

* Create a new post.
* Add a block image (doesn't need to have any actual image).
* Click Publish. 
* The expected behavior is that the pre-publish panel includes both the Tags and the PostFormat components to prompt the user to add them. Both should be collapsed. See:

| Closed | Opened |
| --- | --- |
| ![pre-publish-panel-copy-long-closed](https://user-images.githubusercontent.com/583546/43855589-6768d142-9b46-11e8-9954-6637279302c5.png) | ![pre-publish-panel-copy-toffumatt](https://user-images.githubusercontent.com/583546/43885487-4ec1f02e-9bb9-11e8-8536-15dc4916b983.png) |

Test variations of the above: with the document sidebar closed (as this is when tags are fetched), with tags and `image` as the current post format, etc. The suggested post format is shown below the current post format, if there is any suggestion - typing `wp.data.select('core/editor').getSuggestedPostFormat()` in the console will give you the same result (or `null` if none is to be suggested).

### TODO

- [x] Make text translatable.
- [x] Try a more conversational post format suggestion (see https://github.com/WordPress/gutenberg/issues/7426#issuecomment-403977765)
- [x] Not shown conditions:
  - [x] Post already has some tag applied.
  - [x] It's a page, not a post (more generally, if the post type allows tags).
  - [x] Don't crash if tags weren't fetched yet.
  - [x] Panel is hidden after adding one tag in the pre-publish panel.

### Open questions 

- [x] Should we suggest only tags or any available taxonomy? We're going to go with tags as a starter.
